gasstation/gasstattion.go

```diff
@@ -22,14 +23,16 @@ import (
 type GasStation struct {
 	bc  blockchain.Blockchain
 	sf  factory.Factory
+	dao blockdao.BlockDAO
```

Comment:
let's define an interface here:
```
type BlockDAO interface {
  GetBlockHash,
  GetBlockByHeight,
}
```

Reply:
similarly, we can define an interface
```
type ExecutionSimulator interface {
  SimulateExecution,
}
```
or pass in the function SimulateExecution as a parameter of NewGasStation

Reply:
fixed in latest commit